### PR TITLE
Store: Add parent category selection to the management form

### PR DESCRIPTION
**Pull request by justinshreve**
This PR adds the ability to select a parent category when updating or creating a new product category. It uses Calypso's `term-tree-selector` component.

Parent branch: #21347.
See: #20295.

To Test:
* Boot up this branch and go to `http://calypso.localhost:3000/store/products/categories/:site`
* Select a category that is a parent category
* Make sure that the "top level category" checkbox is shown
* Uncheck it, and make sure that the category selector appears
* Verify that you can select category as parent
* Go back to the list and select a sub-category
* Verify that its parent is marked in the selector component
* Go back once more to the list, and click "Add new category". Make sure the component works in the create context as well.

screen shot 2018-01-08 at 4 30 34 pm

Note: Saving has not been hooked up yet.

### CODE REVIEW

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 1]
```diff
@@ -218,6 +262,29 @@ class ProductCategoryForm extends Component {
 							</FormFieldSet>
 						</div>
 					</div>
+					<FormFieldSet>
+						<FormLabel>
+							<FormCheckbox checked={ this.state.isTopLevel } onChange={ this.onTopLevelChange } />
+							<span>{ translate( 'Top level category' ) }</span>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/product-categories/form.js]: Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Top level category', { context: 'Categories: New category being created is top level i.e. has no parent'} )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 2]
```diff
@@ -218,6 +262,29 @@ class ProductCategoryForm extends Component {
...
+						</FormLabel>
+
+						{ ! isTopLevel && (
+							<div>
+								<FormLabel>{ translate( 'Select a parent category' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/product-categories/form.js]: Hi! I've found a possible matching string that has already been translated **100** times:
`translate( 'Parent Category' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-product-cat-parent-field)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by justinshreve** [comment]
@jameskoster Thanks. Made a few fixes and changes, and moved the UI to align with the title and description.

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 3]
```diff
@@ -216,6 +260,34 @@ class ProductCategoryForm extends Component {
...
+									<FormCheckbox
+										checked={ this.state.isTopLevel }
+										onChange={ this.onTopLevelChange }
+									/>
+									<span>{ translate( 'Top level category' ) }</span>
+								</FormLabel>
+
+								{ ! isTopLevel && (
+									<div>
+										<FormLabel>{ translate( 'Select a parent category' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/product-categories/form.js]: Hi! I've found a possible matching string that has already been translated **100** times:
`translate( 'Parent Category' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-product-cat-parent-field)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/product-categories/form.js [hunk 4]
```diff
@@ -216,6 +260,34 @@ class ProductCategoryForm extends Component {
 								<FormLabel htmlFor="description">{ translate( 'Description' ) }</FormLabel>
 								{ this.renderTinyMCE() }
 							</FormFieldSet>
+							<FormFieldSet>
+								<FormLabel>
+									<FormCheckbox
+										checked={ this.state.isTopLevel }
+										onChange={ this.onTopLevelChange }
+									/>
+									<span>{ translate( 'Top level category' ) }</span>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/product-categories/form.js]: Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Top level category', { context: 'Categories: New category being created is top level i.e. has no parent'} )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-product-cat-parent-field)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
Works well. Just an overall comment on the UI: It looks just like products, which is good for consistency, but may be a bit too close and could be initially confused for adding a product. 

The Product layout generally matches the front end layout of products. Let's do the same for Categories and reduce the relative size of the image section by about half. This would also bring it a bit more inline with Jay's original design. Like: 

screen shot 2018-01-10 at 2 03 45 pm

**Comment by justinshreve** [comment]
Good call @kellychoffman. I just pushed a change and it looks much better:

screen shot 2018-01-10 at 1 25 18 pm

**Comment by timmyc** [comment]
+1 on that change, was just coming to note that on narrow viewports, the mce toolbar obscures the first line of text in the description:

hidden-text

**Comment by timmyc** [comment]
Re-tested the above with the latest changes - it indeed is looking much better, but as I resize the window a bit some interesting things happen in more narrow viewports:

image-breakpoint

Also noticed this too just now - HTML is showing up in the description on the listing page:

html-description-shown

**Comment by timmyc** [comment]
Not sure if you want to work on those items here or in a follow-up PR - worth noting the same screen resize oddness with the term-tree-selector I believe exists in the editor too when you resize the window a bunch - or at least it did at one point.

**Comment by justinshreve** [comment]
The description/HTML bug is handled in the saving PR.

**Comment by justinshreve** [comment]
Width will be fixed in a follow-up.

File: client/extensions/woocommerce/app/product-categories/update.js [hunk 5]
```diff
@@ -76,20 +88,73 @@ class ProductCategoryUpdate extends React.Component {
...
+		accept( areYouSure, function( accepted ) {
+			if ( ! accepted ) {
+				return;
+			}
+			const successAction = () => {
+				debounce( () => {
+					page.redirect( getLink( '/store/products/categories/:site/', site ) );
+				}, 2000 )();
+				return successNotice(
+					translate( '%(name)s successfully deleted.', {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/product-categories/update.js]: Hi! I've found a possible matching string that has already been translated **76** times:
`translate( '%s was successfully deleted.', { context: 'theme'} )` *ES Score: 7*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-product-cat-parent-field)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**